### PR TITLE
fix-issue-constraint-lsm

### PR DESCRIPTION
**Pull request by FloLey**
# Description

fix issue with INMANTA_LSM_MODULE_CONSTRAINTS that weren't set correctly.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Review by sanderr — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]

File: tests/utils.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 def add_version_constraint_to_project(project_dir: py.path.local):
     constraints = os.environ.get("INMANTA_LSM_MODULE_CONSTRAINTS", "")
-    if constraints:
+    if constraints and os.path.exists(project_dir / "module.yml"):
```
[the remark this answers is not in the document]
Comment (FloLey) [reply tests/utils.py]: I think it is done now but not sure how to test it works :p

[the remark this answers is not in the document]
Comment (sanderr) [reply tests/utils.py]: I would just try it manually: add a breakpoint in a test case that installs a v2 module, set the env var with a constraint, run the test and check the installed version.

File: tests/utils.py [hunk 2]
```diff
@@ -24,11 +24,15 @@
...
-        with open(project_dir / "module.yml", "w") as fh:
-            yaml.dump(module_config, fh)
+        if os.path.exists(project_dir / "module.yml"):
+            with open(project_dir / "module.yml", "r") as fh:
+                module_config = yaml.safe_load(fh)
+                module_config["requires"] = constraints.split(";")
+            with open(project_dir / "module.yml", "w") as fh:
+                yaml.dump(module_config, fh)
+        else:
+            with open(project_dir / "requirements.txt", "a") as fh:
```
Comment (sanderr) [inline tests/utils.py:34]: Oh, this won't work, my bad. V2 modules ignore `requirements.txt`, we need to add it to `setup.cfg`.

**Review by sanderr — CHANGES_REQUESTED** [review]

File: tests/utils.py [hunk 3]
```diff
@@ -24,15 +25,24 @@
...
                 yaml.dump(module_config, fh)
-        else:
-            with open(project_dir / "requirements.txt", "a") as fh:
-                fh.write("\n".join(constraints.split(";")))
+        else:  # v2 module
+            parser = ConfigParser()
+            parser.read(str(project_dir / "setup.cfg"))
+            existing_requires = parser.get("options", "install_requires", fallback="").strip()
+            if existing_requires:
+                existing_requires = existing_requires + "\n"
```
Comment (sanderr) [inline tests/utils.py:39]: It's a string, not a list?

**Review by sanderr — APPROVED** [review]

File: tests/utils.py [hunk 4]
```diff
@@ -24,15 +25,24 @@
...
+            constraints_list = [f"inmanta-module-{c}" for c in constraints.split(";")]
+            parser.set("options", "install_requires", "\n" + existing_requires + "\n".join(constraints_list))
+            with open(project_dir / "setup.cfg", "w") as fh:
+                parser.write(fh)
+            with open(project_dir / "setup.cfg", "w") as fh:
+                parser.write(fh)
```
[the remark this answers is not in the document]
Comment (FloLey) [reply tests/utils.py:45]: oups, don't know how this happened.

File: tests/utils.py (hunk 3 above)
Comment (FloLey) [reply tests/utils.py:39]: yes it's a string; i'll add the typing

**Pull request merged**